convert2rhel/unit_tests/toolopts_test.py

```diff
@@ -145,3 +146,254 @@ def test_both_disable_submgr_and_no_rhsm_options_work(argv, raise_exception, no_
+
+
+@pytest.mark.parametrize(
+    ("argv", "content", "message", "output"),
+    (
+        (
+            mock_cli_arguments(["--password", "pass", "-f", "file", "--config-file", "file"]),
+            ("pass_file", "[subscription_manager]\nactivation_key = pass_cnf_file"),
+            "Password file take precedence over the config file.",
+            {"password": "pass_file", "activation_key": None},
+        ),
+    ),
+)
+def test_multiple_auth_src_files(argv, content, message, output, caplog, monkeypatch, tmp_path):
+    """Test combination of password file, config file and CLI."""
+    path0 = os.path.join(str(tmp_path), "convert2rhel.password")
+    with open(path0, "w") as file:  # pylint: disable=unspecified-encoding
+        file.write(content[0])
+    path1 = os.path.join(str(tmp_path), "convert2rhel.ini")
+    with open(path1, "w") as file:  # pylint: disable=unspecified-encoding
+        file.write(content[1])
+    # Set the paths
+    argv[-3] = path0
+    argv[-1] = path1
+    os.chmod(path1, 0o600)
+
+    monkeypatch.setattr(sys, "argv", argv)
+    monkeypatch.setattr(convert2rhel.toolopts, "CONFIG_PATHS", value=[""])
+    convert2rhel.toolopts.CLI()
+
+    assert message in caplog.text
+    assert convert2rhel.toolopts.tool_opts.activation_key == output["activation_key"]
+    assert convert2rhel.toolopts.tool_opts.password == output["password"]
+
+
+@pytest.mark.parametrize(
+    ("argv", "message", "output"),
+    (
+        (
+            mock_cli_arguments(["--password", "pass", "--activationkey", "key"]),
+            "Set only one of password or activation key.",
+            {"password": "pass", "activation_key": "key"},
+        ),
+    ),
+)
+def test_multiple_auth_src_cli(argv, message, output, caplog, monkeypatch):
+    """Test both auth methods in CLI."""
+    monkeypatch.setattr(sys, "argv", argv)
+    monkeypatch.setattr(convert2rhel.toolopts, "CONFIG_PATHS", value=[""])
+    convert2rhel.toolopts.CLI()
+
+    assert message in caplog.text
+    assert convert2rhel.toolopts.tool_opts.activation_key == output["activation_key"]
+    assert convert2rhel.toolopts.tool_opts.password == output["password"]
+
+
+@pytest.mark.parametrize(
+    ("content", "output"),
+    (
+        (
+            "[subscription_manager]\npassword = correct_password",
+            {"password": "correct_password", "activation_key": None},
+        ),
+        (
+            "[subscription_manager]\nactivation_key = correct_key\nPassword = correct_password",
+            {"password": "correct_password", "activation_key": "correct_key"},
+        ),
+        (
+            "[subscription_manager]\nincorrect_option = incorrect_content",
+            {"password": None, "activation_key": None},
+        ),
+        ("[INVALID_HEADER]\nactivation_key = correct_key", {"password": None, "activation_key": None}),
+        ("nonexisting", {"password": None, "activation_key": None}),
```

Comment:
You could change this "nonexisting" to `None`, it would make it simpler to compare later on in the test: 
```suggestion
        (None, {"password": None, "activation_key": None}),
```